frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ConnectionSection.tsx

```diff
@@ -309,17 +325,25 @@ const NewConnectionField: React.FC<NewConnectionFieldProps> = ({
           setValidations((prev) => ({ ...prev, [field.envVar]: isValid }))
         }
       />
-      {selectedConnectionType && !isUriConnectionType(selectedConnectionType) && (
-        <DataConnectionFolderPathField
-          folderPath={data.storage.path}
-          setFolderPath={(path) => setData('storage', { ...data.storage, path })}
-        />
-      )}
+      {selectedConnectionType &&
+        getConnectionTypeModelServingCompatibleTypes(selectedConnectionType)[0] ===
+          ModelServingCompatibleTypes.S3ObjectStorage && (
```

Comment:
Is it maybe worth factoring out a function so you could do something like:
```ts
isModelServingConnectionType(selectedConnectionType, ModelServingCompatibleTypes.S3ObjectStorage)
```
(not sure on the best name here or whether this is worth it)